commit f9901adf536c8430f29bddb36cc648e02143b8ec
Author: Eric Dumazet <edumazet@google.com>
Date:   Wed Aug 30 09:29:31 2017 -0700

    kcm: do not attach PF_KCM sockets to avoid deadlock
    
    
    [ Upstream commit 351050ecd6523374b370341cc29fe61e2201556b ]
    
    syzkaller had no problem to trigger a deadlock, attaching a KCM socket
    to another one (or itself). (original syzkaller report was a very
    confusing lockdep splat during a sendmsg())
    
    It seems KCM claims to only support TCP, but no enforcement is done,
    so we might need to add additional checks.
    
    Fixes: ab7ac4eb9832 ("kcm: Kernel Connection Multiplexor module")
    Signed-off-by: Eric Dumazet <edumazet@google.com>
    Reported-by: Dmitry Vyukov <dvyukov@google.com>
    Acked-by: Tom Herbert <tom@quantonium.net>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/net/kcm/kcmsock.c b/net/kcm/kcmsock.c
index deca20fb2ce2..0ddcb209bea6 100644
--- a/net/kcm/kcmsock.c
+++ b/net/kcm/kcmsock.c
@@ -1383,6 +1383,10 @@ static int kcm_attach(struct socket *sock, struct socket *csock,
 	if (!csk)
 		return -EINVAL;
 
+	/* We must prevent loops or risk deadlock ! */
+	if (csk->sk_family == PF_KCM)
+		return -EOPNOTSUPP;
+
 	psock = kmem_cache_zalloc(kcm_psockp, GFP_KERNEL);
 	if (!psock)
 		return -ENOMEM;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1020_f9901adf536c8430f29bddb36cc648e02143b8ec.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1020_f9901adf536c8430f29bddb36cc648e02143b8ec.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
